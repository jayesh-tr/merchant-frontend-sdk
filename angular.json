{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "projects",
  "projects": {
    "ng9-business-console": {
      "projectType": "library",
      "root": "projects/ng9-business-console",
      "sourceRoot": "projects/ng9-business-console/src",
      "prefix": "ng9-business-console",
      "schematics": {
        "@schematics/angular:component": {
          "styleext": "scss"
        }
      },
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-ng-packagr:build",
          "options": {
            "tsConfig": "projects/ng9-business-console/tsconfig.lib.json",
            "project": "projects/ng9-business-console/ng-package.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "projects/ng9-business-console/tsconfig.lib.prod.json"
            }
          }
        },
        "test": {
          "builder": "@angular-devkit/build-angular:karma",
          "options": {
            "main": "projects/ng9-business-console/src/test.ts",
            "tsConfig": "projects/ng9-business-console/tsconfig.spec.json",
            "karmaConfig": "projects/ng9-business-console/karma.conf.js"
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": [
              "projects/ng9-business-console/tsconfig.lib.json",
              "projects/ng9-business-console/tsconfig.spec.json"
            ],
            "exclude": [
              "**/node_modules/**"
            ]
          }
        }
      }
    }
  },
  "defaultProject": "ng9-business-console",
  "schematics": {
    "@schematics/angular:component": {
      "styleext": "scss"
    }
  }
}