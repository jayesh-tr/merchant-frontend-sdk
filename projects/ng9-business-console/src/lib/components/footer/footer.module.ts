import { NgModule, ModuleWithProviders } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MaterialModule } from '../../material.module';
import { FooterService } from '../../services/footer/footer.service';
import { FooterComponent } from './footer.component';

@NgModule({
  declarations: [FooterComponent],
  imports: [CommonModule, MaterialModule],
  exports: [FooterComponent],
  providers: [FooterService],
})
export class FooterModule {}
