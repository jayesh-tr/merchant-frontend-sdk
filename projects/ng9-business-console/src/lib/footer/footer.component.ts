import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'ng9-business-console-footer',
  templateUrl: './footer.component.html',
  styleUrls: ['./footer.component.scss']
})
export class FooterComponent implements OnInit {
  title = 'footer component works!';
  constructor() { }

  ngOnInit(): void {
  }

}
