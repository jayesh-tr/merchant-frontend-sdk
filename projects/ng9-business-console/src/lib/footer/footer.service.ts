import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root',
})
export class FooterService {
  constructor() {}

  add(...params: number[]): number {
    let result = 0;
    for (const val of params) {
      result += val;
    }
    return result;
  }
}
