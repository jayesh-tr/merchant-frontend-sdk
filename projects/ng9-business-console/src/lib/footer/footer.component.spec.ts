// import { async, ComponentFixture, TestBed } from '@angular/core/testing';

// import { FooterComponent } from './footer.component';

// describe('FooterComponent', () => {
//   let component: FooterComponent;
//   let fixture: ComponentFixture<FooterComponent>;

//   beforeEach(async(() => {
//     TestBed.configureTestingModule({
//       declarations: [ FooterComponent ]
//     })
//     .compileComponents();
//   }));

//   beforeEach(() => {
//     fixture = TestBed.createComponent(FooterComponent);
//     component = fixture.componentInstance;
//     fixture.detectChanges();
//   });

//   it('should create', () => {
//     expect(component).toBeTruthy();
//   });
// });

import { render } from '@testing-library/angular';
import { FooterComponent } from './footer.component';

describe('FooterComponent', () => {
  it('should render the component', async () => {
    const { getByText } = await render(FooterComponent);
    expect(getByText('footer component works!'));
  });
});
