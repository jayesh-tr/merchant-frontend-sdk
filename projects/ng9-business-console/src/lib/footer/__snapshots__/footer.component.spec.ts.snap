// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterComponent should create 1`] = `
<ng9-business-console-footer
  title={[Function String]}
>
  <h3>
    
  </h3>
</ng9-business-console-footer>
`;
